Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -191,33 +154,32 @@ + (void)setAppKey:(NSString *)appKey {
     [BNCPreferenceHelper writeObjectToDefaults:KEY_APP_KEY value:appKey];
 }
 
-+ (NSString *)getBranchKey {
-    if (!Branch_Key) {
-        Branch_Key = [BNCPreferenceHelper getBranchKey:YES];
-    }
-    
-    return Branch_Key;
-}
-
 + (NSString *)getBranchKey:(BOOL)isLive {
-    NSString *key = nil;
+    BNCPreferenceHelper *instance = [BNCPreferenceHelper getInstance];
+    NSString *key = instance.branchKey;
+    
+    if (key && isLive == instance.isUsingLiveKey) {
+        return key;
+    }
     
-    id ret = [[[NSBundle mainBundle] infoDictionary] objectForKey:KEY_BRANCH_KEY];
+    id ret = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"branch_key"];
```

Comment:
I'm not sure. It felt weird that the BNCPreferences defined that constant, and that they were just kinda strewn about all over. Maybe I'll make a constants file for that and some of the other items.